fedimint-core/src/encoding/mod.rs

```diff
@@ -341,95 +341,6 @@ macro_rules! impl_encode_decode_tuple {
     );
 }
 
-/// Specialized version of Encodable for bytes
-pub fn consensus_encode_bytes<W: std::io::Write>(
-    bytes: &[u8],
-    writer: &mut W,
-) -> Result<usize, Error> {
-    let mut len = 0;
-    len += (bytes.len() as u64).consensus_encode(writer)?;
-    writer.write_all(bytes)?;
-    len += bytes.len();
-    Ok(len)
-}
-
-/// Specialized version of Encodable for static byte arrays
-pub fn consensus_encode_bytes_static<const N: usize, W: std::io::Write>(
-    bytes: &[u8; N],
-    writer: &mut W,
-) -> Result<usize, Error> {
-    writer.write_all(bytes)?;
-    Ok(bytes.len())
-}
-
-struct ReadBytesFromFiniteReaderOpts {
-    len: usize,
-    chunk_size: usize,
-}
-
-/// Read `opts.len` bytes from reader, where `opts.len` could potentially be
-/// malicious.
-///
-/// Adapted from <https://github.com/rust-bitcoin/rust-bitcoin/blob/e2b9555070d9357fb552e56085fb6fb3f0274560/bitcoin/src/consensus/encode.rs#L659>
-#[inline]
-fn read_bytes_from_finite_reader<D: Read + ?Sized>(
-    d: &mut D,
-    mut opts: ReadBytesFromFiniteReaderOpts,
-) -> Result<Vec<u8>, io::Error> {
-    let mut ret = vec![];
-
-    assert_ne!(opts.chunk_size, 0);
-
-    while opts.len > 0 {
-        let chunk_start = ret.len();
-        let chunk_size = core::cmp::min(opts.len, opts.chunk_size);
-        let chunk_end = chunk_start + chunk_size;
-        ret.resize(chunk_end, 0u8);
-        d.read_exact(&mut ret[chunk_start..chunk_end])?;
-        opts.len -= chunk_size;
-    }
-
-    Ok(ret)
-}
-
-/// Specialized version of Decodable for bytes
-pub fn consensus_decode_bytes<D: std::io::Read>(r: &mut D) -> Result<Vec<u8>, DecodeError> {
-    consensus_decode_bytes_from_finite_reader(&mut r.take(MAX_DECODE_SIZE as u64))
-}
-
-/// Specialized version of Decodable for bytes
-pub fn consensus_decode_bytes_from_finite_reader<D: std::io::Read>(
-    r: &mut D,
-) -> Result<Vec<u8>, DecodeError> {
-    let len = u64::consensus_decode_from_finite_reader(r, &ModuleRegistry::default())?;
-
-    let len: usize =
-        usize::try_from(len).map_err(|_| DecodeError::from_str("size exceeds memory"))?;
-
-    let opts = ReadBytesFromFiniteReaderOpts {
-        len,
-        chunk_size: 64 * 1024,
-    };
-
-    read_bytes_from_finite_reader(r, opts).map_err(DecodeError::from_err)
-}
-
-/// Specialized version of Decodable for fixed-size byte arrays
-pub fn consensus_decode_bytes_static<const N: usize, D: std::io::Read>(
```

Comment:
Note for reviewer: This was removed, not inlined. It was not previously used anywhere.